src/Aspire.Hosting/build/Aspire.Hosting.targets

```diff
@@ -34,12 +34,24 @@
     <ItemGroup>
       <AspireProjectMetadataSource Update="@(AspireProjectMetadataSource)">
         <Source>
-        <![CDATA[namespace Projects%3B
-
+        <![CDATA[//------------------------------------------------------------------------------
+// <auto-generated>
+//     This code was generated by a tool.
+//
+//     Changes to this file may cause incorrect behavior and will be lost if
+//     the code is regenerated.
+// </auto-generated>
+//------------------------------------------------------------------------------
+
+namespace Projects%3B
+
+[global::System.CodeDom.Compiler.GeneratedCode("Aspire.Hosting", "]]>$(MicrosoftDotnetSdkInternalPackageVersion)<![CDATA[")]
```

Comment:
`$(MicrosoftDotnetSdkInternalPackageVersion)` is not correct. It is not the Aspire version, it is the version of the .NET SDK we are using to build the workload.

Reply:
Note also, `$(MicrosoftDotnetSdkInternalPackageVersion)` is only set during our local build. It isn't going to be set during a customer's build.

Reply:
I have added this to `Aspire.Hosting.csproj`:
```xml
<ItemGroup>
  <None Include="**/*.props;**/*.targets" ... PerformTextReplacement="True" />
</ItemGroup>
...
<ItemGroup>
  <!-- Copy the package version into Aspire.Hosting.targets,
       this is required to correctly version generated code. -->
  <TextReplacementValue Include="Version" NewValue="$(Version)" />
</ItemGroup>
```

And changed `Aspire.Hosting.targets` like this:
```xml
<AspireVersion>@Version@</AspireVersion>
<AspireVersion Condition="$(AspireVersion) == '@Version@'">$(Version)</AspireVersion>
```

Is this enough to copy the files?

I am sorry for the amount of questions and help needed, but I find the network of `targets`-, `prop-` and `csproj`-files a bit confusing.
I truly appreciate you taking the time to help me :)

Reply:
> I find the network of targets-, prop- and csproj-files a bit confusing.

I found this book to be extremely useful for learning about MSBuild. https://www.amazon.com/Inside-Microsoft-Build-Engine-Foundation/dp/0735645248. The first 3 chapters especially. https://learn.microsoft.com/en-us/visualstudio/msbuild/build-process-overview does a pretty good job too.

Reply:
This book is on MS Library - I'll shoot you a link!